modules/solid_mechanics/src/userobjects/CrackFrontDefinition.C

```diff
@@ -70,7 +71,11 @@ CrackFrontDefinition::CrackFrontDefinition(const InputParameters & parameters) :
       mooseError("q_function_rings not supported with crack_front_points");
   }
   else if (isParamValid("boundary"))
+  {
     _geom_definition_method = CRACK_FRONT_NODES;
+    if (parameters.isParamSetByUser("closed_loop"))
+      mooseError("In CrackFrontDefinition, if 'boundary' is defined, 'closed_loop' should not be set by user!");
+  }
```

Comment:
For the case that user defines the crack front points instead of crack front boundary, this parameter has to be set by the user.